service/backend/pchain/backend.go

```diff
@@ -0,0 +1,110 @@
+package pchain
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto"
+	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	"github.com/coinbase/rosetta-sdk-go/types"
+
+	"github.com/ava-labs/avalanche-rosetta/client"
+	pmapper "github.com/ava-labs/avalanche-rosetta/mapper/pchain"
+	"github.com/ava-labs/avalanche-rosetta/service"
+	"github.com/ava-labs/avalanche-rosetta/service/backend/pchain/indexer"
+)
+
+var (
+	_ service.ConstructionBackend = &Backend{}
+	_ service.NetworkBackend      = &Backend{}
+	_ service.AccountBackend      = &Backend{}
+	_ service.BlockBackend        = &Backend{}
+)
+
+type Backend struct {
+	fac                    *crypto.FactorySECP256K1R
+	networkIdentifier      *types.NetworkIdentifier
+	pClient                client.PChainClient
+	indexerParser          indexer.Parser
+	getUTXOsPageSize       uint32
+	codec                  codec.Manager
+	codecVersion           uint16
+	genesisBlock           *indexer.ParsedGenesisBlock
+	genesisBlockIdentifier *types.BlockIdentifier
+	chainIDs               map[string]string
+	avaxAssetID            ids.ID
+}
+
+func NewBackend(
+	pClient client.PChainClient,
+	indexerParser indexer.Parser,
+	assetID ids.ID,
+	networkIdentifier *types.NetworkIdentifier,
+) *Backend {
+	return &Backend{
+		fac:               &crypto.FactorySECP256K1R{},
+		networkIdentifier: networkIdentifier,
+		pClient:           pClient,
+		getUTXOsPageSize:  1024,
+		codec:             blocks.Codec,
+		codecVersion:      0,
```

Comment:
nit: maybe use `codecVersion: txs.Version,` rather than hardcoding 0, as we do in avalanchego codebase

Reply:
Is it safe to assume `txs.Version` will always match the `blocks.Codec`? I was hesitant to use it as they are exposed in different packages.

Reply:
@bilgehansahin  thanks for the explanation.
TL;DR it is safe.
Let me provide some context. We have done quite some repackaging and cleanup of P-chain codebase which resulted (among other things) into splitting transactions and blocks into their own sub-packages.
The platformvm package's  codec is "blocks.Codec", its version is txs.Version. We may clean this up and introduce a blocks.Version variable to make more readable. Keep you posted on this